plumbing/transport/server/loader.go

```diff
@@ -0,0 +1,51 @@
+package server
+
+import (
+	"gopkg.in/src-d/go-git.v4/plumbing/storer"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport"
+	"gopkg.in/src-d/go-git.v4/storage/filesystem"
+
+	"srcd.works/go-billy.v1"
+	"srcd.works/go-billy.v1/os"
+)
+
+// DefaultLoader is a filesystem loader ignoring host and resolving paths to /.
+var DefaultLoader = NewFilesystemLoader(os.New("/"))
+
+// Loader loads repository's storer.Storer based on an optional host and a path.
+type Loader interface {
+	// Load loads a storer.Storer given a transport.Endpoint.
+	// Returns transport.ErrRepositoryNotFound if the repository does not
+	// exist.
+	Load(ep transport.Endpoint) (storer.Storer, error)
+}
+
+type fsLoader struct {
+	base billy.Filesystem
+}
+
+// NewFilesystemLoader creates a Loader that ignores host and resolves paths
+// with a given base filesystem.
+func NewFilesystemLoader(base billy.Filesystem) Loader {
+	return &fsLoader{base}
+}
+
+func (l *fsLoader) Load(ep transport.Endpoint) (storer.Storer, error) {
+	fs := l.base.Dir(ep.Path)
+	if _, err := fs.Stat("config"); err != nil {
+		return nil, transport.ErrRepositoryNotFound
+	}
+
+	return filesystem.NewStorage(fs)
+}
+
+type MapLoader map[transport.Endpoint]storer.Storer
```

Comment:
As discussed before, we'll proceed with current naming. We agreed that it's correct that `Loader` is in the `server` package, since there are no uses out of it. We can revisit this if a use case comes up for it.

With respect to naming of both `Loader` and `Storer`, the discussion is still open, but we'll not stop the PR.